examples/text_labels_and_annotations/custom_legends.py

```diff
@@ -0,0 +1,56 @@
+"""
+========================
+Composing Custom Legends
+========================
+
+Composing custom legends piece-by-piece.
+
+Sometimes you don't want a legend that is explicitly tied to data that
+you have plotted. For example, say you have plotted 10 lines, but don't
+want a legend item to show up for each one. If you simply plot the lines
+and call ``ax.legend()``, you will get the following:
+"""
+# sphinx_gallery_thumbnail_number 2
+from matplotlib import rcParams, cycler
+import matplotlib.pyplot as plt
+import numpy as np
+
+N = 10
+data = [np.logspace(0, 1, 100) + np.random.randn(100) + ii for ii in range(N)]
+data = np.array(data).T
+cmap = plt.cm.coolwarm
+rcParams['axes.prop_cycle'] = cycler(color=cmap(np.linspace(0, 1, N)))
+
+fig, ax = plt.subplots()
+lines = ax.plot(data)
+ax.legend(lines, loc=(1.05, .05), )
+
+##############################################################################
+# In this case, we can compose a legend using Matplotlib objects that aren't
+# explicitly tied to the data that was plotted. For example:
+
+from matplotlib.lines import Line2D
+custom_lines = [Line2D([0], [0], color='b', lw=4),
+                Line2D([0], [0], color='grey', lw=4),
+                Line2D([0], [0], color='r', lw=4)]
+
+fig, ax = plt.subplots()
+lines = ax.plot(data)
+ax.legend(custom_lines, ['Cold', 'Medium', 'Hot'], loc=(1.05, .8),)
+
+
+###############################################################################
+# There are many other Matplotlib objects that can be used in this way. Below
+# is a list of common ones.
+
+from matplotlib.patches import Patch
+from matplotlib.lines import Line2D
+
+legend_elements = {'Line': Line2D([0], [0], color='b', lw=4),
+                   'Scatter': Line2D([0], [0], marker='o', color='w',
+                                     markerfacecolor='g', markersize=15),
+                   'Color Patch': Patch(facecolor='orange', edgecolor='r')}
```

Comment:
More a note for the future: if #8292 is some day merged in (I hope so!), it would be nice to remember updating this example to include one case or two.